extensions/markdown-language-features/src/features/workspaceSymbolProvider.ts

```diff
@@ -81,8 +90,23 @@ class VSCodeWorkspaceMarkdownDocumentProvider extends Disposable implements Work
 	}
 
 	private async getMarkdownDocument(resource: vscode.Uri): Promise<SkinnyTextDocument | undefined> {
-		const doc = await vscode.workspace.openTextDocument(resource);
-		return doc && isMarkdownFile(doc) ? doc : undefined;
+		let result = this._docIndex.getByUri(resource);
+		if (result) {
+			return await vscode.workspace.openTextDocument(resource);
+		}
+
+		let bytes = await vscode.workspace.fs.readFile(resource);
+		// We assume that markdown is in UTF-8
+		let text = new StringDecoder('utf-8').write(Buffer.from(bytes));
```

Comment:
I suggest using `Buffer.toString(encoding)` and skipping the dependency.

Reply:
Could you explain what do you mean? I can't find Buffer.toString(encoding)

Reply:
It's a node api: https://nodejs.org/api/buffer.html#buffer_buf_tostring_encoding_start_end

Reply:
@mjbvz Ok. Thanks. Fixed.